src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            if (!string.IsNullOrEmpty(autoLogonCountValue))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_AutoLogonCount"));
+            }           
+            
+            return warningReasons;
+        }
+
+        private void UpdateMachineSpecificRegistrySettings(string domainName, string userName)
+        {
+            var hive = RegistryHive.LocalMachine;
+
+            // SetValue(RegistryHive targetHive, string subKeyName, string name, string value)
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName, userName);
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName, domainName);
+
+            //this call is to take the backup of the password key if already exists as we delete the key in the next step
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword, "");
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword);
+            
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount, "");
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount);
+
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon, "1");
+
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI, "0");
+
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption, "");
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText, "");
+        }
+
+        private void UpdateUserSpecificRegistrySettings(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                UpdateRegistrySettingsForDifferentUser(securityId);                
+            }
+            else
+            {
+                UpdateRegistrySettingsForCurrentUser();
+            }
+        }
+
+        private void UpdateRegistrySettingsForCurrentUser()
+        {
+            var hive = RegistryHive.CurrentUser;
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaver, RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver, "0");
```

Comment:
DomainPolicy controls the actual value which is stored in other setting. After syncing up with Prachi, we decided not to alter the domain policies. So now we are touching only one setting.

---
In reply to: [118825754](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118825754) [](ancestors = 118825754)